internal/bzlmod/go_deps.bzl

```diff
@@ -643,8 +644,12 @@ def _go_deps_impl(module_ctx):
 
             go_repository_args.update(repo_args)
 
+        if "version" in go_repository_args:
+            resolved_go_modules[go_repository_args["importpath"]] = go_repository_args["version"]
+
         go_repository(**go_repository_args)
 
+    module_ctx.file("./resolved_go_modules.json", content=json.encode_indent(resolved_go_modules))
```

Comment:
Files written by module extensions are emitted into an unspecified temporary directory that should be removed after execution. We would need to instantiate a repo with this info or accept an absolute path via an environment variable.

Reply:
Makes sense, I'm propagating this similar to config.json used by rules_go. I was able to verify that I can access this file as `@bazel_gazelle_go_repository_config//:resolved_deps.json`. Would this approach work?

Reply:
I can also expose this via a separate repo rule if that's preferable. On a related note, while I can pull this into my workspace as `@bazel_gazelle_go_repository_config//:resolved_deps.json`, `bazel mod tidy` still thinks this repo is an indirect dependency and removes it from `MODULE.bazel` (I'm explicitly passing it in as inputs to both a build rule and a `sh_binary` rule). Is that perhaps a bug? If not, can I add a directive to inform `bazel mod tidy` that I intend to keep that use_repo call?

Reply:
Unsure why Windows build of rules_go is failing but interestingly I see Bazel warn that the repo is an indirect dependency in Gazelle's own MODULE.bazel file:
```
(16:03:24) WARNING: C:/b/bk-windows-xmds/bazel/bazel-gazelle/tests/bcr/go_work/MODULE.bazel:27:24: The module extension go_deps defined in @gazelle//:extensions.bzl reported incorrect imports of repositories via use_repo():
--
  |  
  | Imported, but reported as indirect dependencies by the extension:
  | bazel_gazelle_go_repository_config, org_golang_google_protobuf
```

Reply:
Looks like the failure was intermittent and unrelated to the change